src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package vic.cinema.dao.impl;
+
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import vic.cinema.dao.MovieDao;
+import vic.cinema.exeptions.DataProcessingException;
+import vic.cinema.lib.Dao;
+import vic.cinema.model.Movie;
+import vic.cinema.util.HibernateUtil;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
please see checklist: Use try with finally, where you use transaction

Reply:
Done